src/AuthModule/components/Register/Register.tsx

```diff
@@ -1,7 +1,322 @@
+  const togglePasswordVisibility = (): void => {
+    setShowPassword((prevState: boolean) => !prevState);
+  };
+  const togglePasswordVisibility2 = (): void => {
+    setShowPassword2((prevState: boolean) => !prevState);
+  };
+  // Custom validation function to check if passwords match
+  const validatePasswordMatch = (value: unknown) => {
+    const password = watch("password"); // Get the value of the "password" field
+    return value === password || "Passwords do not match"; // Return error message if passwords don't match
+  };
+  // Custom validation function to check if name match
+  const validateUserName = (value: string) => {
+    // Regular expression to match characters followed by numbers without spaces
+    const regex = /^[a-zA-Z]+[0-9]+$/;
+    // Test the value against the regular expression
+    const isValid = regex.test(value);
+    // Return true if the value matches the pattern, otherwise return false
+    return (
+      isValid ||
+      "Username must contain characters and end with numbers without spaces"
+    );
+  };
+  // change data to FormData 
+
+  const appendToFormData = (data :FormDataRegister) => {
+    const profileImage = data.profileImage[0];
+    const formData = new FormData();
+    formData.append("userName", data.userName);
+    formData.append("email", data.email);
+    formData.append("country", data.country);
+    formData.append("phoneNumber", data.phoneNumber);
+    formData.append("password", data.password);
+    formData.append("confirmPassword", data.confirmPassword);
+    formData.append("profileImage", profileImage);
+    return formData ;
+  };
+  // senD Data to Api
+  
+  const onSubmit = async (data: FormDataRegister) => {
+    setSpinner(true);
+     const fotmData = appendToFormData(data)
+    try {
+      const response = await axios.post(
+        "https://upskilling-egypt.com:3003/api/v1/Users/Register",
+        fotmData
+      );
+    
+      console.log(response);
+      showSuccessToast(response.data.message);
+      navigate("/VerifyEmail")
+    } catch (error) {
+      if (axios.isAxiosError(error) && error.response) {
+        showErrorToast(error.response.data.message)}
+        else {
+          // Handle other types of errors here
+          showErrorToast("An error occurred.");
+        }
+    } finally {
+      setSpinner(false);
+    }
+
+  };
+
   return (
-    <div>Register</div>
-  )
+    <div className="Auth-container vh-100 row align-items-center justify-content-center overflow-auto gx-0 flex-nowrap ">
+      <div className="logo  col-md-5 text-center  ">
+        <img src={logo} alt="logo" className="mb-3" />
+      </div>
+
+      <div className="login-container  col-md-8 rounded-4 px-3 py-2">
+        <div className=" p-2">
+          <p className="text-white">Welcome To PMS!</p>
+          <h3 className="color-text mb-3">Create New Account</h3>
+        </div>
+
+        <form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
It is preferable to navigate the form from  left top right and then go down, but by using the tap key for navigation, it moves from the first input and finishes the first column and then moves to the second column which doesn't have a good user experience.